src/CommunityToolkit.Maui.Core/Interfaces/IFolderPicker.shared.cs

```diff
@@ -0,0 +1,24 @@
+using CommunityToolkit.Maui.Core.Primitives;
+
+namespace CommunityToolkit.Maui.Core;
+
+/// <summary>
+/// Allows picking a folder from the file system.
+/// </summary>
+public interface IFolderPicker
+{
+	/// <summary>
+	/// Allows picking a folder from the file system.
+	/// </summary>
+	/// <param name="initialPath">Initial path</param>
+	/// <param name="cancellationToken"><see cref="CancellationToken"/></param>
+	/// <returns><see cref="Folder"/></returns>
+	Task<Folder?> PickAsync(string initialPath, CancellationToken cancellationToken);
```

Comment:
Let's return `Task<Folder>` (non-nullable) and throw an Exception if the folder cannot be returned.

This aligns more closely to .NET MAUI's `FilePicker` that does things like `throws throw new PlatformNotSupportedException`

For us, we can leverage `FileNotFoundException` when appropriate.